packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+>**Note:** The examples in this guide use Swift 5, SwiftUI, and a target of iOS 15 or later. The SwiftUI portions pass state information using multiple bindings, consider using more robust ways of sharing and updating data in production apps. For more information, see the [Apple developer documentation for state and data flow in SwiftUI](https://developer.apple.com/documentation/swiftui/state-and-data-flow/).
+
+The code in this guide manages the sign-on flow using a singleton class (`SignInController`). The class handles asynchronous responses from the SDK by implementing the delegate functions of the `IDXClientDelegate` protocol. The SDK also provides completion handlers and Swift Concurrency (`async/await`).
+
+### 1: Display the initial sign-in UI
+
+Enable the user to start the sign-in flow, such as presenting a Sign In button or a form with fields for a username and password (you won't use those values until step 3 when you [authenticate the user credentials](#_3-authenticate-the-user-credentials)).
+
+### 2: Start the sign-on flow
+
+Initialize the SDK client and request the initial server response after the user starts the sign-in process.
+
+This code also sets the caller as the delegate for asynchronous callbacks from the SDK:
+
+```swift
+// Initialize the SDK client using a Configuration object loaded previously.
+IDXClient.start(with: configuration) { (client, error) in
+    guard let client = client else {
+		self.finish(with: error)
+		return
+	}
+
+	self.client = client
+
+    // Set this object as the delegate for asynchronous calls.
+	client.delegate = self
+
+	// Request the first response from the Okta org.
+	client.resume()
+}
+```
+
+The `configuration` (`IDXClient.IDXClientConfiguration`) parameter in the call to `IDX.start(with:)` contains the configuration information for your Okta org, and is usually initialized from an `Okta.plist` file. The call to `client.resume()` requests the first response of the sign-sequence.
+
+### 3: Authenticate the user credentials
+
+The SDK uses a `Response` object to represent the server request for the current sign-on step. It contains a collection of remediations (`Remedation`) that specify the type of step, any general messages, and a nested collection of forms (`Remedation.Form`) containing fields (`Remedation.Form.Field`) that represent the various elements of the UI, such as an entry field with a label. Completing a step usually requires setting the value of one or more fields. The remediation is then sent back to the server with the updated values by calling `proceed()`. The server processes the values and then sends the next step.
+
+The kind of sign-on step is represented by the `type` property of the remediation. In this step the server is requesting the username and password, an `.identify` remediation type.
```

Comment:
I agree with all but the recommendation for the first sentence. A good catch that .identify is the subject of the second sentence.

For the first one, the sentence is about the kind of sign-on rather, so that would make it the subject. For active voice the subject comes first. Similarly, from a flow/story perspective, the reader has the understanding of "sign-on step" at this point, but not it's representation. This structure connects those two concepts together starting with the one that's already top-of-mind for the reader (which also is how things work at a neuronal level, but that's a different story ;-)). Putting type first is likely to require the reader to go through the sentence at least twice.

How about:

> The kind of sign-on step is represented by the `type` property of the remediation. In this step, the remediation type `.identify` indicates the server's requesting the user's name and password.

Reply:
That's fine. I was trying to make the verbactive rather than passive - something that Susan is pretty hot on.